pdr_backend/predictoor/base_predictoor_agent.py

```diff
@@ -163,3 +238,6 @@ def get_prediction(
         timestamp: int,  # pylint: disable=unused-argument
     ) -> Tuple[bool, float]:
         pass
+
+def _tx_failed(tx):
+    return tx is None or tx["status"] != 1
```

Comment:
This avoided me having to do the test of "tx is None" and "tx["status"]" in three separate places.

BTW you always need `is None` not `== None`.